File: Front/GestionHeuresSup/src/app/Models/stage-pfe.ts
/* tslint:disable */
import {Departement} from './departement';
import {Enseignant} from './enseignant';
import {EtudiantPFE} from './etudiant-pfe';
import {Statut} from './statut';

export class StagePFE {
  public id: number;
  public sujet: string;
  public societe: string;
  public departement: Departement;
  public enseignant: Enseignant;
  public etudiantPFE1: EtudiantPFE;
  public etudiantPFE2: EtudiantPFE;
  public statut: Statut;


  constructor(id?: number, sujet?: string, societe?: string, departement?: Departement, enseignant?: Enseignant, etudiantPFE1?: EtudiantPFE, etudiantPFE2?: EtudiantPFE, statut?: Statut) {
    this.id = id;
    this.sujet = sujet;
    this.societe = societe;
    this.departement = departement;
    this.enseignant = enseignant;
    this.etudiantPFE1 = etudiantPFE1;
    this.etudiantPFE2 = etudiantPFE2;
    this.statut = statut;
  }


  getId(): number {
    return this.id;
  }

  setId(value: number) {
    this.id = value;
  }

  getSujet(): string {
    return this.sujet;
  }

  setSujet(value: string) {
    this.sujet = value;
  }

  getSociete(): string {
    return this.societe;
  }

  setSociete(value: string) {
    this.societe = value;
  }

  getDepartement(): Departement {
    return this.departement;
  }

  setDepartement(value: Departement) {
    this.departement = value;
  }

  getEnseignant(): Enseignant {
    return this.enseignant;
  }

  setEnseignant(value: Enseignant) {
    this.enseignant = value;
  }

  getEtudiantPFE1(): EtudiantPFE {
    return this.etudiantPFE1;
  }

  setEtudiantPFE1(value: EtudiantPFE) {
    this.etudiantPFE1 = value;
  }

  getEtudiantPFE2(): EtudiantPFE {
    return this.etudiantPFE2;
  }

  setEtudiantPFE2(value: EtudiantPFE) {
    this.etudiantPFE2 = value;
  }

  getStatut(): Statut {
    return this.statut;
  }

  setStatut(value: Statut) {
    this.statut = value;
  }
}

File: Front/GestionHeuresSup/src/main/java/com/iset/projetPFE/entites/StagePFE.java
package com.iset.projetPFE.entites;

import java.io.Serializable;

import javax.annotation.Generated;
import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import javax.persistence.OneToOne;

@Entity
public class StagePFE implements Serializable{
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private int id;
	private String sujet;
	private String societe;
	@ManyToOne(fetch = FetchType.EAGER)
	private Departement departement;
	@ManyToOne(fetch = FetchType.EAGER)
	private Enseignant enseignant;
	@OneToOne(fetch = FetchType.EAGER,cascade = CascadeType.MERGE)
	private EtudiantPFE etudiantPFE1;
	@OneToOne(fetch = FetchType.EAGER ,cascade = CascadeType.MERGE)
	
	private EtudiantPFE etudiantPFE2;
	@ManyToOne(fetch =FetchType.EAGER)
	private Statut statut;

	public int getId() {
		return id;
	}
	public void setId(int id) {
		this.id = id;
	}
	public String getSujet() {
		return sujet;
	}
	public void setSujet(String sujet) {
		this.sujet = sujet;
	}
	public String getSociete() {
		return societe;
	}
	public void setSociete(String societe) {
		this.societe = societe;
	}
	public Departement getDepartement() {
		return departement;
	}
	public void setDepartement(Departement departement) {
		this.departement = departement;
	}
	public Enseignant getEnseignant() {
		return enseignant;
	}
	public void setEnseignant(Enseignant enseignant) {
		this.enseignant = enseignant;
	}
	public EtudiantPFE getEtudiantPFE1() {
		return etudiantPFE1;
	}
	public void setEtudiantPFE1(EtudiantPFE etudiantPFE1) {
		this.etudiantPFE1 = etudiantPFE1;
	}
	public EtudiantPFE getEtudiantPFE2() {
		return etudiantPFE2;
	}
	public void setEtudiantPFE2(EtudiantPFE etudiantPFE2) {
		this.etudiantPFE2 = etudiantPFE2;
	}
	public Statut getStatut() {
		return statut;
	}
	public void setStatut(Statut statut) {
		this.statut = statut;
	}
	@Override
	public String toString() {
		return "StagePFE [id=" + id + ", sujet=" + sujet + ", societe=" + societe + "]";
	}
	
	
}
